src/azure-cli/azure/cli/command_modules/resource/_bicep.py

```diff
@@ -8,23 +8,49 @@
 import stat
 import platform
 import subprocess
+import json
 
-from pathlib import Path
+from json.decoder import JSONDecodeError
 from contextlib import suppress
+from datetime import datetime, timedelta
 
 import requests
 import semver
 
 from six.moves.urllib.request import urlopen
 from knack.log import get_logger
-from azure.cli.core.azclierror import FileOperationError, ValidationError, UnclassifiedUserFault, ClientRequestError
+from azure.cli.core.api import get_config_dir
+from azure.cli.core.azclierror import (
+    FileOperationError,
+    ValidationError,
+    UnclassifiedUserFault,
+    ClientRequestError,
+    InvalidTemplateError,
+)
 
 # See: https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
 _semver_pattern = r"(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)(?:-(?P<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+(?P<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?"  # pylint: disable=line-too-long
+
+# See: https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-syntax#template-format
+_template_schema_pattern = r"https?://schema\.management\.azure\.com/schemas/[0-9a-zA-Z-]+/(?P<templateType>[a-zA-Z]+)Template\.json#?"  # pylint: disable=line-too-long
+
+_config_dir = get_config_dir()
+_bicep_installation_dir = os.path.join(_config_dir, "bin")
+_bicep_version_check_file_path = os.path.join(_config_dir, "bicepVersionCheck.json")
+_bicep_version_check_cache_ttl = timedelta(minutes=10)
```

Comment:
May I ask why the ttl of this cache is 10 minutes?

Reply:
Setting ttl to 10 minutes means it will sent at most 6 requests per hour, which is good enough to avoid GitHub throttling for anonymous requests (60 / hour / IP address). Other other hand, we do want the bicep commands to detect version upgrade of Bicep CLI as soon as we do a release, and I feel like 10 minute of delay is acceptable.

Reply:
This is still a workaround, though. CI/CD runs that depend on Bicep can still get throttled. We are also thinking about setting up CDN for caching the Bicep releases to completely eliminate the throttling issue.

Reply:
Got it, thanks~